### ISSUE TITLE
Initialization of monit returns AssertException: Cannot open /dev/null -- Bad file descriptor error

### ISSUE THREAD

**Issue by boyonwheels** [saltstack/salt#32190]
### Description of Issue/Question

If you try to execute any module that interacts with monit (e.g. service.start or monit.status or cmd.run "monit status", etc) from the Salt master to a minion you get the following error:

 AssertException: Cannot open /dev/null -- Bad file descriptor
     raised in init_env at src/env.c:92 

These errors do not occur if you use salt-call directly on the minion.
### Setup

This is a test/default setup of Saltstack on FreeBSD 10.2 (master) and FreeBSD 10.1 (minion). I believe this is reproducible on other system versions, too, but have not tested.
### Steps to Reproduce Issue

Try to interact with monit using any module that does not provide stdin. For example, this is broken:

% sudo salt www-test cmd.run "monit status"  
www-test:
    AssertException: Cannot open /dev/null -- Bad file descriptor
     raised in init_env at src/env.c:92
ERROR: Minions returned with non-zero exit code

However, this works:

% sudo salt www-test cmd.run "monit status" stdin=""
www-test:
    The Monit daemon 5.16 uptime: 2d 21h 14m 
    [...additional monit output removed...]

The same errors show up if you try to use Saltstack's built-in service and monit modules.
### Versions Report

MASTER:
Salt Version:
           Salt: 2015.8.7

Dependency Versions:
         Jinja2: 2.8
       M2Crypto: 0.24.0
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
         Python: 2.7.11 (default, Mar 17 2016, 01:24:00)
           RAET: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.4
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.0
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist:  
        machine: amd64
        release: 10.2-RELEASE-p14

MINION:
Salt Version:
           Salt: 2015.8.7

Dependency Versions:
         Jinja2: 2.8
       M2Crypto: 0.22
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
         Python: 2.7.11 (default, Feb 20 2016, 01:20:42)
           RAET: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.2
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.3
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist:  
        machine: amd64
        release: 10.1-RELEASE-p16

**Comment by boyonwheels** [comment]
I also created a bug with the Monit folks with related information. https://bitbucket.org/tildeslash/monit/issues/337/assertexception-cannot-open-dev-null-bad

**Comment by jfindlay** [comment]
@boyonwheels, thanks for reporting.  This seems to be an upstream bug, but we could implement the workaround you have demonstrated.

**Comment by boyonwheels** [comment]
Thanks for the quick reply, @jfindlay! Some additional notes from today (unfortunately, not enough time to try too many things just yet):

I decided to try to use cmd.run in a state with stdin but that didn't fix the issue like running the execution module directly. Afterwards, I noticed that the state documentation didn't list the stdin option. Although, putting it in the state file didn't cause an error, either. Is this expected?

The workaround I think I've decided on (to try to work with Monit's behavior) is to remap stdin to /dev/null:

monit_service_hack:
  service.enabled:
    - name: monit
  cmd.run:
    - name: "0<>/dev/null /usr/sbin/service monit status || 0<>/dev/null /usr/sbin/service monit start"

That seems reasonable to me as a workaround (although I'm open to other ideas!).
